test/functional/lua/vim_spec.lua

```diff
@@ -1273,6 +1273,53 @@ describe('lua stdlib', function()
       eq("", result[8])
     end)
 
+    it('should allow options to be reset to default values', function()
+      local result = exec_lua [[
+        local result = {}
+        table.insert(result, vim.opt.shiftwidth:default())
+        vim.opt.shiftwidth = 42
+        table.insert(result, vim.opt.shiftwidth:get())
+        vim.opt.shiftwidth:reset()
+        table.insert(result, vim.opt.shiftwidth:get())
+        return result
+      ]]
+
+      eq(8, result[1])
+      eq(42, result[2])
+      eq(8, result[3])
+    end)
+
+    it('should allow local options to be set to their global value', function()
+      local result =  exec_lua [[
+        local result = {}
+        vim.opt.shiftwidth = 8
+        vim.opt_local.shiftwidth = 4
+        table.insert(result, vim.opt_local.shiftwidth:get())
+        vim.opt_local.shiftwidth = vim.opt_global.shiftwidth:get()
```

Comment:
does this test what you thought it was going to? Shouldn't this be `reset()` or something?

Reply:
`:reset()` resets to default value, not global value. To set a local value to a global value you do it explicitly like this. Personally I find this much easier to reason about than all of the different `:set<`, `:setlocal<` variants that Vim script uses, which I've had to look up about a dozen different times while working on this PR >_>